src/main/java/Duke.java

```diff
@@ -1,10 +1,230 @@
+                }
+            }
+        } catch (DukeException e) {
+            printTextWithDivider(e.getMessage());
+        }
+    }
+
+    private static void listCommand(String[] inputs) throws DukeException {
+        if (inputs.length == 2) {
+            throw new DukeException("OOPS!!! I'm sorry, but I don't know what that means :-(.\n");
+        }
+
+        StringBuilder str = new StringBuilder();
+        str.append("Here are the tasks in your list:\n");
+        for (int i = 0; i < taskList.size(); i++) {
+            // Display task as 1-index
+            str.append(i + 1).append(".").append(taskList.get(i)).append("\n");
+        }
+        printTextWithDivider(str.toString());
+    }
+
+    private static void addTaskCommand(TaskType taskType, String[] inputs) {
+        if (inputs.length == 1 || inputs[1].equals("")) {
+            throw new DukeException("OOPS!!! The description of " + taskType.getTaskType() + " cannot be empty.\n");
+        }
+
+        Task task;
+        switch (taskType) {
+            case TODO: {
+                task = new Todo(inputs[1]);
+                break;
+            }
+            case DEADLINE: {
+                String[] deadlineInputs = inputs[1].split("/by", 2);
+
+                if (deadlineInputs.length == 1 || deadlineInputs[1].equals("")) {
+                    throw new DukeException("OOPS!!! The date of a deadline cannot be empty.\n");
+                }
+                task = new Deadline(deadlineInputs[0], deadlineInputs[1]);
+                break;
+            }
+            case EVENT: {
+                String[] eventInputs = inputs[1].split("/at", 2);
+
+                if (eventInputs.length == 1 || eventInputs[1].equals("")) {
+                    throw new DukeException("OOPS!!! The date and time of an event cannot be empty.\n");
+                }
+                task = new Event(eventInputs[0], eventInputs[1]);
+                break;
+            }
+            default: {
+                throw new DukeException("OOPS!!! Invalid task type.\n");
+            }
+        }
+
+        taskList.add(task);
+
+        storage.appendTaskToFile(task);
+
+        String addTaskMessage =  "Got it. I've added this task:\n" +
+                "  " + task + "\n" +
+                "Now you have " + taskList.size() + " task(s) in the list.\n";
+
+        printTextWithDivider(addTaskMessage);
+    }
+
+    private static void markTaskCommand(String[] inputs) throws DukeException {
+        if (inputs.length == 1 || inputs[1].equals("")) {
+            throw new DukeException("OOPS!!! The task index cannot be empty.\n");
+        }
+        try {
+            // Tasks are stored as 0-index but display as 1-index
+            // Minus 1 to get the correct task in the taskList
+            int taskIndex = Integer.parseInt(inputs[1]) - 1;
+            Task task = taskList.get(taskIndex);
+            task.markAsDone();
+            storage.writeAllTasksToFile(taskList);
+
```

Comment:
I'm not sure I like how the code here aims to do multiple things all at once. From carrying out the task to printing to the user interface. Do you think it is a good idea to maybe extract some of similar logic to separate classes?